cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -586,19 +594,24 @@ private ModelCompletenessRequirements modelCompletenessRequirements(Collection<G
    * @param goalNames Goal names (and empty list of names indicates all goals).
    */
   public boolean meetCompletenessRequirements(List<String> goalNames) {
-    Collection<Goal> goals = goalsByPriority(goalNames).values();
+    Collection<Goal> goals = goalsByPriority(goalNames, false).values();
     MetadataClient.ClusterAndGeneration clusterAndGeneration = _loadMonitor.refreshClusterAndGeneration();
     return goals.stream().allMatch(g -> _loadMonitor.meetCompletenessRequirements(
         clusterAndGeneration, g.clusterModelCompletenessRequirements()));
   }
 
   /**
    * Get a goals by priority based on the goal list.
+   *
+   * @param goals A list of goals.
+   * @param skipHardGoalCheck True if the provided {@code goals} do not have to contain all hard goals, false otherwise.
+   * @return A map of goal priority to goal.
    */
-  private Map<Integer, Goal> goalsByPriority(List<String> goals) {
+  private Map<Integer, Goal> goalsByPriority(List<String> goals, boolean skipHardGoalCheck) {
```

Comment:
Since in some cases we hard-code the value for `skipHardGoalCheck` when calling this function, I am wondering whether it would make the code more clear if we keep the functionality of this function the same as before, and create a separate function for sanity checking the existence of hard goals in the given goals.

Reply:
current logic is populate the `goal list` with `default goal list` if it is null, then do the check(which is guarantee to pass).